### PR TITLE
Migrate Vue-2 to Vue-3 for FLINT-UI

### PR DESCRIPTION
**Pull request by HarshCasper**
## Resources

- GitHub: https://github.com/vuejs/core/tree/main/packages/vue-compat
- Blog: https://www.vuemastery.com/blog/vue-3-migration-build/
- Official reference: https://v3-migration.vuejs.org/

### CODE REVIEW

**Comment by Palaksharma23** [comment]
image

Little UI issues are there in GCBM and GCBM is not working fine.

**Comment by SanjaySinghRajpoot** [comment]
> image
> 
> Little UI issues are there in GCBM and GCBM is not working fine.

Yes, I am having the same issue as above. Moreover, I am unable to see the toggle button below the GCBM heading.  I don't know whether I am only facing this issue or if it's with everyone. Along with I got some warnings with `:v-deep` tag used in the slider component when I first compiled the code. Later they disappeared from the terminal after 2-3 compile reloads

Have a look at this for the above warning https://stackoverflow.com/questions/48032006/how-do-i-use-deep-or-or-v-deep-in-vue-js

Else the changes look to me.

**Comment by Janvi-Thakkar** [comment]
> > image
> > Little UI issues are there in GCBM and GCBM is not working fine.
> 
> Yes, I am having the same issue as above. Moreover, I am unable to see the toggle button below the GCBM heading. I don't know whether I am only facing this issue or if it's with everyone. Along with I got some warnings with `:v-deep` tag used in the slider component when I first compiled the code. Later they disappeared from the terminal after 2-3 compile reloads
> 
> Have a look at this for the above warning https://stackoverflow.com/questions/48032006/how-do-i-use-deep-or-or-v-deep-in-vue-js
> 
> Else the changes look to me.

I am not even able to see the toggle button neither on the deployed link nor on my local machine.

File: flint.ui/src/components/FileUpload/FileUpload.vue [hunk 1]
```diff
@@ -1,54 +1,87 @@
...
+      <div
+        v-if="!files.db.length"
         id="dropzoneDB"
-        ref="myVueDropzoneDB"
-        :options="dropzoneOptionsDB"
-        :use-custom-slot="true"
-        @vdropzone-removed-file="fileRemoved"
-        @vdropzone-file-added="fileAddedDB"
+        class="dropzone-gcbm"
+        :style="{ border: resultDB.isDragActive.value ? '2px dashed #0087ff' : '2px dashed #bdbdbd' }"
```
Comment (sohamsshah) [inline flint.ui/src/components/FileUpload/FileUpload.vue:36]: Can we replace the inline styling with design system styling or a css class for this case?

File: flint.ui/src/components/FileUpload/FileUpload.vue [hunk 2]
```diff
@@ -1,54 +1,87 @@
...
         </div>
-      </vue-dropzone>
+      </div>
+      <div v-else class="files-container">
+        <div v-for="(file, index) in files.db" :key="'file-gcbm-db-' + index" class="file-container">
+          <span class="text-lg text-blueGray-700">
+            {{ file.name }}
+          </span>
+          <br />
+          <span>{{ parseFloat((file.size / 1000).toFixed(2)) + ' KB' }}</span>
```
Comment (sohamsshah) [inline flint.ui/src/components/FileUpload/FileUpload.vue:52]: Abstract this logic into a function in this file or a separate /utils folder

File: flint.ui/src/components/Stepper/StepperGCBM.vue [hunk 3]
```diff
@@ -22,36 +22,36 @@ export default {
...
-     } else if (current == 2) {
-       this.$router.push({
-           name: "gcbmrun",
-           params: { current },
-       })
-     }
-    },
-  },
-};
+      if (current == 0) {
```
Comment (sohamsshah) [inline flint.ui/src/components/Stepper/StepperGCBM.vue:27]: Small nitpick:
-  We can use switch-case here
- Add comments for explaining what does `0`, `1` etc. numbers signify

**Review by sohamsshah — CHANGES_REQUESTED** [review]
Great work @YashKandalkar ! 👏
Have requested a few minor changes, please incorporate them.

Thanks for the PR!

**Comment by YashKandalkar** [comment]
@Palaksharma23 @Janvi-Thakkar @SanjaySinghRajpoot @sohamsshah Thanks for the review! I have made the requested changes, PTAL

**Comment by Janvi-Thakkar** [comment]
> @Palaksharma23 @Janvi-Thakkar @SanjaySinghRajpoot @sohamsshah Thanks for the review! I have made the requested changes, PTAL

Great work @YashKandalkar :) The toggle button is still not visible on deployed preview in my machine not sure whether it is only happening with me or if anyone else has the same issue

**Pull request merged**